src/backend/common/SerialQueue.h

```diff
@@ -110,12 +110,14 @@ namespace backend {
 
     template<typename T>
     void SerialQueue<T>::Enqueue(const std::vector<T>& values, Serial serial) {
+        ASSERT(values.size() > 0);
```

Comment:
Hm, I thought it would be strange to call Enqueue but not actually insert anything into the queue. You would be associating nothing with some serial value.

But regardless it breaks if you enqueue an empty vector because in `T& SerialQueue<T>::Iterator::operator*() const`
```
if (serialIterator == nullptr) {
    return *storageIterator->second.begin();
}
```
`storageIterator->second` is empty.

We could just modify the dereference, or even more simply just do nothing if you enqueue an empty vector?